packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -498,6 +498,57 @@ export namespace externalAppAuthentication {
     });
   }
 
+  /**
+   * @beta
+   * @hidden
+   * API to authenticate power platform connector plugins
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param titleId ID of the acquisition
+   * @param pluginId pluginId of the connector plugin
+   * @param signInUrl signInUrl for the connctor page listing the connector. This is optional
+   * @param oauthWindowParameters parameters for the signIn window
+   * @returns A promise that resolves when authentication succeeds and rejects with InvokeError on failure
+   */
+  export function authenticateWithPowerPlatformConnectorPlugins(
+    titleId: string,
+    pluginId: string,
+    signInUrl?: string,
```

Comment:
>string [](http://example.com/codeflow?start=16&length=6)

Can we accept this as a `URL` type instead of a `string` type?

Reply:
We shouldn't be consistent with bad design just for the sake of consistency.

I'm not sure who "we" is regarding enforcing the rule but I can guarantee you that every PR I've ever done where I see URLs being passed as strings has received this comment. We should always use the most appropriate type to hold the data otherwise we might as well write directly in JavaScript.

I'll sign off on this because I don't know what other constraints under which you are operating and I don't want to hold up the whole effort for this. Feel free to discuss the tradeoffs further with @erinha if needed.

Reply:
I agree that we should stop perpetuating bad habits. I am opening a work item to track modifying these private APIs to accept url parameters as `URL`s instead of strings.

@ashishsingh2105 do you know of any reason why you can't pass url as a URL type in the code that calls this API?